templates/python/hgcBiasedGenProcesses_cfi.py

```diff
@@ -0,0 +1,150 @@
+                                                                             'MDME(207,1) = 0    !W decay into mu+ nu_mu', 
+                                                                             'MDME(208,1) = 0    !W decay into tau+ nu_tau'),
+                                             # This is a vector of ParameterSet names to be read, in this order
+                                             parameterSets = cms.vstring('pythiaUESettings', 
+                                                                         'processParameters')
+                                         )
+                                     )
+
+    #ttbar
+    if proc=='ttbar':
+        addJetFilter=True
+        process.generator = cms.EDFilter("Pythia8GeneratorFilter",
+                                         maxEventsToPrint = cms.untracked.int32(1),
+                                         pythiaPylistVerbosity = cms.untracked.int32(1),
+                                         filterEfficiency = cms.untracked.double(1.0),
+                                         pythiaHepMCVerbosity = cms.untracked.bool(False),
+                                         comEnergy = cms.double(14000.0),                                         
+                                         crossSection = cms.untracked.double(1),                                         
+                                         PythiaParameters = cms.PSet(
+                                             pythia8CommonSettingsBlock,
+                                             pythia8CUEP8M1SettingsBlock,
+                                             processParameters = cms.vstring(
+                                                 'Top:gg2ttbar    = on',
+                                                 'Top:qqbar2ttbar = on',
+                                                 '6:m0 = 172.5',    # top mass',
+                                             ),
+                                             parameterSets = cms.vstring('pythia8CommonSettings',
+                                                                         'pythia8CUEP8M1Settings',
+                                                                         'processParameters',
+                                                                     )                                             
+                                         )
+                                     )
+
+def defineJetBasedBias(process,jetColl="ak8GenJetsNoNu",thr=100,minObj=1):
+
+    """
+    defines a path to execute to filter the output based on a given gen-jet collection
+    and the presence of minObj-jets with pt>thr
+    """
+
+    #gen level selection of gen jets in the endcap        
+    setattr(process,'ee'+jetColl,cms.EDFilter("CandViewShallowCloneProducer",
+                                              src = cms.InputTag(jetColl),
+                                              cut = cms.string("abs(eta)<3.0 && abs(eta)>1.5") ) )
```

Comment:
We use |eta| > 1.479 as default for the particle gun. I think it would be good to be consistent.